pimp-it.lua

```diff
@@ -0,0 +1,1060 @@
+        end
+    end
+    return true
+end
+
+--Apply only after previously validating
+function ApplyProfession(dwf, profession, min, max)
+    local prof = cloned.professions[profession]
+    --todo: implement persistent profession counting
+    --prof.cur = prof.cur + 1
+    for skill, bonus in pairs(prof.skills) do
+        local sTable = GetSkillTable(dwf, skill)
+        if sTable == nil then
+            utils.insert_or_update(dwf.status.current_soul.skills, { new = true, id = df.job_skill[skill], rating = 0 }, 'id')
+            sTable = GetSkillTable(dwf, skill)
+        end
+        local points = rng.rollInt(engineID, min, max)
+        sTable.rating = sTable.rating < points and points or sTable.rating
+        sTable.rating = sTable.rating + bonus
+        sTable.rating = sTable.rating > 20 and 20 or sTable.rating
+        sTable.rating = sTable.rating < 0 and 0 or sTable.rating
+        if args.debug and tonumber(args.debug) >= 2 then print(skill .. ".rating = " .. sTable.rating) end
+    end
+    return true
+end
+
+--Apply only after previously validating
+function ApplyJob(dwf, jobName) --job = dorf_jobs[X]
+    local jd = cloned.distributions[jobName]
+    local job = cloned.jobs[jobName]
+    if args.debug and tonumber(args.debug) >= 3 then print(dwf,job,jobName, PimpData[jobName]) end
+    PimpData[jobName].count = PimpData[jobName].count + 1
+    jd.cur = PimpData[jobName].count
+    local id = tostring(dwf.id)
+    DwarvesData[id] = {}
+    DwarvesData[id]['job'] = jobName
+    DwarvesData[id]['professions'] = {}
+    if not PimpData[jobName] then
+        PimpData[jobName] = {}
+    end
+    dwf.custom_profession = jobName
+    RollStats(dwf, job.types)
+    
+    -- Apply required professions
+    local bAlreadySetProf2 = false
+    local job_req_sequence = rng.num_sequence:new()
+    for i=1,ArrayLength(job.req) do
+        job_req_sequence:add(i)
+    end
+    local gen = rng.crng:new(engineID,false,job_req_sequence)
+    --two required professions are set as the professional titles for a dwarf [prof1, prof2]
+    --so when more than 2 are required it is necessary to randomize the iteration of their application to a dwarf
+    --this is done with rpairs and the above RNG code
+    gen:shuffle()
+    job_req_sequence:add(0) --adding an out of bounds key (ie. 0) to ensure rpairs won't keep going forever
+    --[note it is added after shuffling]
+    local i = 0
+    for _, prof in rpairs(job.req, gen) do
+        --> Set Profession(s) (by #)
+        i = i + 1 --since the key can't tell us what iteration we're on
+        if i == 1 then
+            dwf.profession = df.profession[prof]
+        elseif i == 2 then
+            bAlreadySetProf2 = true
+            dwf.profession2 = df.profession[prof]
+        end
+        --These are required professions for this job class
+        ApplyProfession(dwf, prof, 11, 17)
+    end
+        
+    -- Loop tertiary professions
+    -- Sort loop (asc)
+    --[[]]
```

Comment:
This kind of stuff is hard to parse, even with syntax highlighting. Can you just remove this and the matching ``--]]`` below?

Reply:
Are you asking to remove the comments or the `--[[]]` part?

Reply:
Just the --[[]] (line 459)

Reply:
Okay, looking at it I get what you mean. A disabled commented out code block if you will. It's removed now. For the next push.